airbyte-webapp/src/components/connection/CatalogTree/next/useCatalogTreeTableRowProps.test.tsx

```diff
@@ -0,0 +1,258 @@
+    expect(result.current.statusIcon).toEqual(null);
+    expect(result.current.pillButtonVariant).toEqual("grey");
+  });
+  it("should return disabled styles for a row that starts disabled", () => {
+    // eslint-disable-next-line
+    jest.spyOn(bulkEditService, "useBulkEditSelect").mockImplementation(() => [false, () => null] as any); // not selected for bulk edit
+    jest.spyOn(connectionFormService, "useConnectionFormService").mockImplementation(() => {
+      return {
+        initialValues: {
+          syncCatalog: {
+            streams: [
+              {
+                ...mockInitialValues.syncCatalog?.streams[0],
+                config: { ...mockInitialValues.syncCatalog?.streams[0].config, selected: false },
+              },
+            ],
+          },
+        },
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      } as any;
+    });
+    jest.spyOn(formik, "useField").mockImplementationOnce(() => {
+      // eslint-disable-next-line
+      return [{}, { error: undefined }] as any; // no error
+    });
+    const { result } = renderHook(() =>
+      useCatalogTreeTableRowProps({
+        ...mockStream,
+        config: { ...mockStream.config!, selected: false },
+      })
+    );
+
+    expect(result.current.streamHeaderContentStyle).toEqual(classNames(styles.streamHeaderContent, styles.disabled));
+    expect(result.current.statusIcon).toEqual(null);
+    expect(result.current.pillButtonVariant).toEqual("grey");
+  });
+  it("should return added styles for a row that is added", () => {
+    // eslint-disable-next-line
+    jest.spyOn(bulkEditService, "useBulkEditSelect").mockImplementation(() => [false, () => null] as any); // not selected for bulk edit
+    jest.spyOn(connectionFormService, "useConnectionFormService").mockImplementation(() => {
+      return {
+        initialValues: {
+          syncCatalog: {
+            streams: [
+              {
+                ...mockInitialValues.syncCatalog?.streams[0],
+                config: { ...mockInitialValues.syncCatalog?.streams[0].config, selected: false },
+              },
+            ],
+          },
+        },
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      } as any;
+    });
+    jest.spyOn(formik, "useField").mockImplementationOnce(() => {
+      // eslint-disable-next-line
+      return [{}, { error: undefined }] as any; // no error
+    });
+    const { result } = renderHook(() =>
+      useCatalogTreeTableRowProps({
+        ...mockStream,
+        config: { ...mockStream.config!, selected: true }, // selected true
+      })
+    );
+
+    expect(result.current.streamHeaderContentStyle).toEqual(classNames(styles.streamHeaderContent, styles.added));
+    // we care that iconName="plus" and className="icon plus" but it is tricky to test a react node returned from a hook like this
+    expect(result.current.statusIcon).toMatchSnapshot();
```

Comment:
Turns out testing a react node that gets returned from the `renderHook` was a bit complex.  Landed on doing a snapshot test, but including notes about the important pieces that should definitely _not_ change.